### PR TITLE
[REF] transform the setting of defaults in CustomField::create to be like (some) other entities

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Code cleanup only

Before
----------------------------------------
Less consistent / readable

After
----------------------------------------
More consistent / readable

Technical Details
----------------------------------------
This pattern (massaging the params & then copying) is more consistent. PR also moves creating the transaction to the start of the fn

I am cleaning up towards adding a bulk function that can be utilised by api v4 to create multiple fields with some efficiency on
sql statements (which matters when adding 2 or more fields to a large table)

Comments
----------------------------------------

### CODE REVIEW

**Comment by seamuslee001** [comment]
@eileenmcnaughton can you rebase this please

**Comment by colemanw** [comment]
@eileenmcnaughton what happened?

**Comment by eileenmcnaughton** [comment]
I think I mis-pushed it to be empty & it auto-closed

**Comment by eileenmcnaughton** [comment]
it's back!

**Comment by eileenmcnaughton** [comment]
@colemanw passing now

File: CRM/Core/BAO/CustomField.php [hunk 1]
```diff
@@ -2167,6 +2157,24 @@ public static function customOptionGroup($includeFieldIds = NULL) {
...
+  public static function getDefaults() {
+    return [
+      'is_required' => FALSE,
+      'is_searchable' => FALSE,
+      'in_selector' => FALSE,
+      'is_search_range' => FALSE,
+      //CRM-15792 - Custom field gets disabled if is_active not set
+      // this would ideally be a mysql default.
+      'is_active' => TRUE,
+      'is_view' => FALSE,
```
Comment (colemanw) [inline CRM/Core/BAO/CustomField.php]: Why wouldn't all of these be mysql defaults?

Comment (eileenmcnaughton) [reply CRM/Core/BAO/CustomField.php]: Yeah beats me - but I wasn't going that far in this PR

**Pull request merged**